core/domain/email_manager.py

```diff
@@ -455,11 +514,14 @@ def _send_email(
             'Details:\n%s %s\n%s\n\n' %
             (recipient_id, email_subject, cleaned_plaintext_body))
         return
+
     @transaction_services.run_in_transaction_wrapper
-    def _send_email_transactional():
+    def _send_email_transactional() -> None:
         """Sends the email to a single recipient."""
         sender_name_email = '%s <%s>' % (sender_name, sender_email)
 
+        # Ruling out the possibility of None for mypy type checking.
+        assert recipient_email is not None
```

Comment:
Should we rather on lines 496 to 496 (where `get_email_from_user_id` is called) create a new variable that will not be optional?

Reply:
Done!